### PR TITLE
Add JavaDocs for Item classes

### PR DESCRIPTION
**Pull request by YHWong20**
Added Javadocs for `Item` classes.

### CODE REVIEW

File: src/main/java/seedu/binbash/item/Item.java [hunk 1]
```diff
@@ -1,44 +1,85 @@
...
+     */
+    protected Item(String itemName, String itemDescription, int itemQuantity, double itemCostPrice) {
         this.itemName = itemName;
         this.itemDescription = itemDescription;
         this.itemQuantity = itemQuantity;
         this.itemCostPrice = itemCostPrice;
         this.totalUnitsPurchased = itemQuantity;
     }
 
+    /**
```
Comment (PureUsagi) [inline src/main/java/seedu/binbash/item/Item.java:24]: Follows JavaDocs formating standard, great work!

File: src/main/java/seedu/binbash/item/PerishableOperationalItem.java [hunk 2]
```diff
@@ -3,6 +3,12 @@
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 
+/**
```
Comment (PureUsagi) [inline src/main/java/seedu/binbash/item/PerishableOperationalItem.java:6]: The JavaDoc is very descriptive!

**Review by PureUsagi — APPROVED** [review]
Great work adding comprehensive JavaDocs!

File: src/main/java/seedu/binbash/item/Item.java [hunk 3]
```diff
@@ -1,44 +1,85 @@
 package seedu.binbash.item;
 
+/**
+ * Abstract Item class, which represents an Item that can be stored in the ItemList (inventory).
+ */
```
Comment (nur-haziq) [inline src/main/java/seedu/binbash/item/Item.java:5]: This is good!

Perhaps we can further clarify that subclasses will extend this class to add specific functionality for different item types.

**Review by nur-haziq — APPROVED** [review]
Clean and well written JavaDocs. 👍

File: src/main/java/seedu/binbash/item/Item.java (hunk 3 above)
Comment (YHWong20) [reply src/main/java/seedu/binbash/item/Item.java:5]: Will do that, thanks!

**Pull request merged**